### PR TITLE
Layering and Error refactor

### PR DESCRIPTION
**Pull request by akshaisarma**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: src/main/java/com/yahoo/bullet/bql/query/LayeredSchema.java [hunk 1]
```diff
@@ -15,82 +15,201 @@
...
+    public FieldLocation findField(String field, int minimumDepth) {
         if (schema == null) {
             // If the schema is null, ignore the subschema and just return Type.UNKNOWN
-            return Type.UNKNOWN;
+            return FieldLocation.from(null, Type.UNKNOWN, depth);
         }
         Type type = schema.getType(field);
-        if (type != Type.NULL) {
-            return type;
+        if (type != Type.NULL && depth >= minimumDepth) {
```
Comment (0aix) [inline src/main/java/com/yahoo/bullet/bql/query/LayeredSchema.java:108]: Could we just skip to the next layer if depth < minimumDepth?

**Review by 0aix — APPROVED** [review]
looks really good 👀

File: src/main/java/com/yahoo/bullet/bql/query/LayeredSchema.java (hunk 1 above)
Comment (akshaisarma) [reply src/main/java/com/yahoo/bullet/bql/query/LayeredSchema.java]: Sure pulled out the depth check

**Comment by akshaisarma** [comment]
Also added a isLocked checked in OrderByProcessor, so it doesn't lock unless it was already locked

**Review by 0aix — APPROVED** [review]

**Pull request merged**